kitchen-tests/cookbooks/end_to_end/recipes/macos.rb

```diff
@@ -54,6 +54,39 @@
   action :enable
 end
 
+### Begin MacOS nonsense
+# What you see below here is a pile of crazy.
+# The End-To-End recipe wants to update git on MacOS to the latest version.
+# MacOS doesn't like this. We get back symlink update errors, et al
+# We then have to use this to take control of enough of the OS
+# to update git without actually hurting anything else. UGH
+
+file "/usr/local/var/homebrew/locks/git@2.35.1.formula.lock" do
+  mode "0777"
+  owner "root"
+end
+
+file "/usr/local/Cellar/git@2.35.1/2.35.1/etc/bash_completion.d/git-completion.bash" do
+  mode "0777"
+  owner "root"
+end
+
+execute "changing ownership of the git cask" do
+  command "chmod -R 777 /usr/local/Cellar/git@2.35.1"
+  live_stream true
+end
+
+execute "Sledge Hammer removal of the offending Git version" do
+  command "rm -rf /usr/local/Cellar/git@2.35.1"
+  live_stream true
+end
```

Comment:
We have a few thoughts from the meeting on this. I know this was merged in master already, but...
1. If this exact version of git isn't installed, Chef will blow up with NoEnclosingDirectory, so this feels incredibly brittle
2. If they move to a new version of git, this fix will not work anymore, I think?
3. We never cleanup the lock file - and if we're gonna leave it around, with 0777 perms, lets be explicit about hte group (wheel?)
4. Do we really want 0777? Maybe 1777?

Reply:
yeah agreed. @johnmccrae Any thoughts?